File: ai-verifier/verifier.py
import openai
import os
from dotenv import load_dotenv
from models import SolutionRequest

load_dotenv()
openai.api_key = os.getenv("OPENAI_API_KEY")

def evaluate_solution(data: SolutionRequest):
    prompt = f"""
    Квест ID: {data.quest_id}
    Решение: {data.solution}

    Проверь решение. Дай фидбэк, score (0-100) и вердикт (pass/fail).
    """
    response = openai.ChatCompletion.create(
        model="gpt-4",
        messages=[{"role": "user", "content": prompt}],
        temperature=0.3
    )
    content = response.choices[0].message.content
    return {"feedback": content}

File: backend/src/verifier.ts
import { OpenAI } from "openai";

const openai = new OpenAI({
    apiKey: process.env.OPENAI_API_KEY || "",
});

export async function verifyTask(task: string, answer: string): Promise<string> {
    const prompt = `Проверь решение задачи "${task}". Ответ: "${answer}". Дай фидбэк и оценку от 1 до 10.`;

    const completion = await openai.chat.completions.create({
        model: "gpt-4",
        messages: [
            { role: "system", content: "Ты ассистент, который проверяет задания по программированию." },
            { role: "user", content: prompt }
        ]
    });

    return completion.choices[0].message?.content || "Нет ответа от модели.";
}
